Clean up build entitlements on watchOS

brentfulgham opened the pull request:
#### 4718ec21fbae6c97ca81df0ea931c39764d87b18
<pre>
Clean up build entitlements on watchOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=255356">https://bugs.webkit.org/show_bug.cgi?id=255356</a>
&lt;rdar://107651745&gt;

Reviewed by Mark Lam.

We do not allow JIT on watchOS, so it&apos;s confusing to include the JIT entitlements
required on iOS on a platform that doesn&apos;t support them. This patch removes them.

* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/262916@main">https://commits.webkit.org/262916@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a000b74169161e4faf20005bf454df1432f3df4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2944 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3009 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3106 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4349 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3353 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3084 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3047 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2579 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2972 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3347 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2633 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4141 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/2971 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/843 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2613 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2475 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2605 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2664 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3886 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3013 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2418 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2646 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2616 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/740 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2634 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2833 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -512,8 +512,7 @@ then
 elif [[ "${WK_PLATFORM_NAME}" == iphoneos ||
-        "${WK_PLATFORM_NAME}" == appletvos ||
-        "${WK_PLATFORM_NAME}" == watchos ]]
+        "${WK_PLATFORM_NAME}" == appletvos ]]
```
Is is possible to check ${WK_PLATFORM_NAME}" == watchOS inside ios_family_process_webcontent_entitlements? If so, perhaps that would be a smaller change?

brentfulgham replied:
Looks like that will work!

MenloDorian reviewed (approved):
r=me

brentfulgham committed:
Clean up build entitlements on watchOS
https://bugs.webkit.org/show_bug.cgi?id=255356
<rdar://107651745>

Reviewed by Mark Lam.

We do not allow JIT on watchOS, so it's confusing to include the JIT entitlements
required on iOS on a platform that doesn't support them. This patch removes them.

* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: https://commits.webkit.org/262916@main
